RecoTracker/TkSeedGenerator/plugins/JetCoreMCtruthSeedGenerator.h

```diff
@@ -0,0 +1,193 @@
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/Common/interface/DetSet.h"
+#include "DataFormats/SiPixelCluster/interface/SiPixelCluster.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "DataFormats/JetReco/interface/Jet.h"
+#include "DataFormats/SiPixelDigi/interface/PixelDigi.h"
+#include "DataFormats/GeometryVector/interface/VectorUtil.h"
+#include "DataFormats/SiPixelDetId/interface/PXBDetId.h"
+#include "DataFormats/Math/interface/Point3D.h"
+#include "DataFormats/Math/interface/Vector3D.h"
+#include "DataFormats/Candidate/interface/Candidate.h"
+
+#include "RecoLocalTracker/ClusterParameterEstimator/interface/PixelClusterParameterEstimator.h"
+#include "RecoLocalTracker/Records/interface/TkPixelCPERecord.h"
+
+#include "SimDataFormats/TrackerDigiSimLink/interface/PixelDigiSimLink.h"
+
+#include "TrackingTools/GeomPropagators/interface/StraightLinePlaneCrossing.h"
+#include "TrackingTools/GeomPropagators/interface/Propagator.h"
+#include "TrackingTools/Records/interface/TrackingComponentsRecord.h"
+
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+
+#include <boost/range.hpp>
+#include <boost/foreach.hpp>
+#include "boost/multi_array.hpp"
+
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+#include "SimDataFormats/Track/interface/SimTrack.h"
+
+#include "SimDataFormats/Vertex/interface/SimVertex.h"
+
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+
+#include "SimDataFormats/TrackingHit/interface/PSimHit.h"
+
+#include "TTree.h"
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+
+namespace edm {
+  class Event;
+  class EventSetup;
+}  // namespace edm
+
+class JetCoreMCtruthSeedGenerator : public edm::one::EDProducer<edm::one::SharedResources> {
+public:
+  explicit JetCoreMCtruthSeedGenerator(const edm::ParameterSet&);
+  ~JetCoreMCtruthSeedGenerator() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  // A pointer to a cluster and a list of tracks on it
+  struct TrackAndState {
+    TrackAndState(const reco::Track* aTrack, TrajectoryStateOnSurface aState) : track(aTrack), state(aState) {}
+    const reco::Track* track;
+    TrajectoryStateOnSurface state;
+  };
+
+  template <typename Cluster>
+  struct ClusterWithTracks {
+    ClusterWithTracks(const Cluster& c) : cluster(&c) {}
+    const Cluster* cluster;
+    std::vector<TrackAndState> tracks;
+  };
+
+  typedef ClusterWithTracks<SiPixelCluster> SiPixelClusterWithTracks;
+
+  typedef boost::sub_range<std::vector<SiPixelClusterWithTracks>> SiPixelClustersWithTracks;
```

Comment:
are these used?

Reply:
only the first one. I remove the second 
https://github.com/cms-sw/cmssw/blob/31ededd5d7a7e720f2bcad3643af8bea1eea20ad/RecoTracker/TkSeedGenerator/plugins/JetCoreMCtruthSeedGenerator.cc#L138